lib/active_merchant/billing/gateways/adyen_checkout.rb

```diff
@@ -220,8 +220,6 @@ def add_shopper_interaction(post, payment, options={})
       end
 
       def add_recurring_processing_model(post, options)
-        return unless options.dig(:stored_credential, :reason_type) || options[:recurring_processing_model]
```

Comment:
How do we know that for any purchase ever we should set the `post[:recurringProcessingModel]` flag?
I know removing this guard solves the PSP case but is it the right solution?

Reply:
OK, I will restore this, and I will pass `recurringProcessingModel` as "Subscription" from the conduit